File: 贪心/392判断子序列/392.判断子序列.java
/*
 * @lc app=leetcode.cn id=392 lang=java
 *
 * [392] 判断子序列
 */

// @lc code=start
class Solution {
    public boolean isSubsequence(String s, String t) {
        int slen = s.length(),tlen = t.length();
        int i = 0, j = 0;
        while(i < slen && j < tlen) {
            if(s.charAt(i) == t.charAt(j)) {
                i++;
            }
            j++;
        }
        return i == slen;
    }
}
// @lc code=end

File: 贪心/392判断子序列/392.判断子序列.js
/*
 * @lc app=leetcode.cn id=392 lang=javascript
 *
 * [392] 判断子序列
 */

// @lc code=start
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
  if(!s.length) return true;
  let index = 0;
  let slen = s.length;
  for(let i = 0; i < t.length; i++) {
      if (t[i] == s[index]) {
          index++;
          if(index == slen) {
              return true;
          }
      }
  }
  return false;
};
// @lc code=end

